### PR TITLE
slim-sidebar right to left support

### PR DESCRIPTION
**Pull request by kaedroho**
This PR adds right to left support for the sidebar. Note that I've kept the page explorer LTR for now:

image

### CODE REVIEW

File: client/src/components/Sidebar/Sidebar.scss [hunk 1]
```diff
@@ -22,11 +22,18 @@
         width: 15px;
         height: 16px;
     }
+
+    [dir='rtl'] & {
+        right: 12px;
```
Comment (fabienheureux) [inline client/src/components/Sidebar/Sidebar.scss:27]: This value could be defined in a css variable

File: client/scss/components/_main-nav.scss [hunk 2]
```diff
@@ -299,16 +299,27 @@ body.explorer-open {
...
+    }
+
     .wrapper,
     body.nav-open .wrapper {
         transform: none;
-        padding-left: $menu-width;
+        padding-left: var(--sidebar-width);
 
         @include transition(padding-left $menu-transition-duration ease);
+        @include transition(padding-right $menu-transition-duration ease);
```
Comment (fabienheureux) [inline client/scss/components/_main-nav.scss:312]:
```suggestion
        @include transition(padding-left $menu-transition-duration ease, padding-right $menu-transition-duration ease);
```

File: client/src/components/Sidebar/menu/MenuItem.scss [hunk 3]
```diff
@@ -78,6 +78,7 @@
 .menuitem-label {
     @include transition(opacity $menu-transition-duration ease);
     margin-left: 0.5em;
+    margin-right: 0.5em;
```
Comment (fabienheureux) [inline client/src/components/Sidebar/menu/MenuItem.scss:81]: This value could be defined in a css variable

File: client/src/components/Sidebar/menu/SubMenuItem.scss [hunk 4]
```diff
@@ -7,18 +11,25 @@
...
         width: 16px;
         height: 16px;
-        transform: translate3d(13px, 0, 0);
+        --translation: 13px;
+    }
+
+    [dir='rtl'] & {
+        --scale: -1;
+        right: unset;
+        left: 15px;
```
Comment (fabienheureux) [inline client/src/components/Sidebar/menu/SubMenuItem.scss:32]: This value could be defined in a css variable

File: client/src/components/Sidebar/modules/Search.scss [hunk 5]
```diff
@@ -53,15 +55,25 @@
...
+            margin-right: var(--svg-margin);
+            transition: margin-right $menu-transition-duration ease, margin-left $menu-transition-duration ease;
         }
 
         .sidebar--slim & {
             width: 100%;
 
             svg {
-                margin-right: 0;
+                --svg-margin: 0;
```
Comment (fabienheureux) [inline client/src/components/Sidebar/modules/Search.scss:66]: Question : this is set initially on `&__submit`, couldn't it be overriden in `.sidebar--slim &` directly ?

File: client/src/components/PageExplorer/PageExplorerPanel.tsx [hunk 6]
```diff
@@ -143,7 +143,7 @@ class PageExplorerPanel extends React.Component<PageExplorerPanelProps, PageExpl
 
     return (
       <Transition name={transition} className="c-page-explorer" component="nav" label={STRINGS.PAGE_EXPLORER}>
-        <div key={depth} className="c-transition-group">
+        <div key={depth} className="c-transition-group" dir="ltr">
```
Comment (fabienheureux) [inline client/src/components/PageExplorer/PageExplorerPanel.tsx:146]: Question : Is it out of the scope of the current PR ? Is there a dedicated issue for to support RTL in the explorer ? 
I tried setting `direction: rtl` on the explorer and it does not appear to be a massive change, I would be happy to work on this if no work has started yet on this issue 

image

**Review by fabienheureux — COMMENTED** [review]
I am not assigned on this issue but saw it was in the UI project board. 
I left a few comments @kaedroho, nothing critical though.

File: client/src/components/PageExplorer/PageExplorerPanel.tsx (hunk 6 above)
Comment (kaedroho) [reply client/src/components/PageExplorer/PageExplorerPanel.tsx:146]: Yep, I left this out for now (or I reckon this PR will be massive!), but there isn't a dedicated issue for this yet

Comment (zerolab) [reply client/src/components/PageExplorer/PageExplorerPanel.tsx:146]: Looks like this PR needs rebasing on `main` after #7844, plus fixing the merge conflicts.

Comment (thibaudcolas) [reply client/src/components/PageExplorer/PageExplorerPanel.tsx:146]: I would expect this to require a lot of re-work, as our preferred approach for RTL support will be to switch to CSS logical properties (https://github.com/wagtail/wagtail/discussions/7793) now that we don’t have support for IE11.

The good news is that switch to logical properties should largely be a matter of search-and-replace.

**Comment by thibaudcolas** [comment]
Superseded by #8051, with the exception of icons mirroring, which I propose to implement for the whole of the admin in one go in #8127.

**Pull request closed without merging**